### ISSUE TITLE
orderasc/orderdesc returns duplicate nodes 

### ISSUE THREAD

**Issue by clemensw** [dgraph-io/dgraph#1918]
If you suspect this could be a bug, follow the template.

- What version of Dgraph are you using?
docker latest: 9a14a4e53776 (v0.9.4)

- Have you tried reproducing the issue with latest release?
yes

- What is the hardware spec (RAM, OS)?
8GB, Ubuntu 17.10 

- Steps to reproduce the issue (command/config used to run Dgraph).
https://tour.dgraph.io/basic/9/ show the use of offset and first with orderasc.  The query with orderasc seems to return each node twice, yielding incorrect results when used with offset:

Query:
{
  michael_friends_first(func: allofterms(name, "Michael")) {
    name
    age
    friend (orderasc: name@.) {
      name@.
    }
  }
}

Response:
{
  "data": {
    "michael_friends_first": [
      {
        "name": "Michael",
        "age": 39,
        "friend": [
          {
            "name@.": "Amit"
          },
          {
            "name@.": "Artyom"
          },
          {
            "name@.": "Catalina"
          },
          {
            "name@.": "Sang Hyun"
          },
          {
            "name@.": "Sarah"
          },
          {
            "name@.": "Artyom"
          },
          {
            "name@.": "Amit"
          },
          {
            "name@.": "Amit"
          },
          {
            "name@.": "Sang Hyun"
          }
        ]
      }
    ]
  },
  "extensions": {
    "server_latency": {
      "parsing_ns": 48003,
      "processing_ns": 3586108,
      "encoding_ns": 994211
    },
    "txn": {
      "start_ts": 51,
      "lin_read": {
        "ids": {
          "1": 6
        }
      }
    }
  }
}

Requesting 'name' instead of 'name@.' limits the output to the two nodes with unqualified strings, however even requesting for name@en yields the duplicates:
Query:
{
  michael_friends_first(func: allofterms(name, "Michael")) {
    name
    age
    friend (orderasc: name) {
      name@en
    }
  }
}

Response:
{
  "data": {
    "michael_friends_first": [
      {
        "name": "Michael",
        "age": 39,
        "friend": [
          {
            "name@en": "Amit"
          },
          {
            "name@en": "Artyom"
          },
          {
            "name@en": "Sang Hyun"
          },
          {
            "name@en": "Artyom"
          },
          {
            "name@en": "Amit"
          },
          {
            "name@en": "Amit"
          },
          {
            "name@en": "Sang Hyun"
          }
        ]
      }
    ]
  },
  "extensions": {
    "server_latency": {
      "parsing_ns": 55604,
      "processing_ns": 3569197,
      "encoding_ns": 877610
    },
    "txn": {
      "start_ts": 58,
      "lin_read": {
        "ids": {
          "1": 6
        }
      }
    }
  }
}

Without orderasc the names are only returned once:
Query:
{
  michael_friends_first(func: allofterms(name, "Michael")) {
    name
    age
    friend {
      name@.
    }
  }
}
Response:
{
  "data": {
    "michael_friends_first": [
      {
        "name": "Michael",
        "age": 39,
        "friend": [
          {
            "name@.": "Amit"
          },
          {
            "name@.": "Sarah"
          },
          {
            "name@.": "Sang Hyun"
          },
          {
            "name@.": "Artyom"
          },
          {
            "name@.": "Catalina"
          }
        ]
      }
    ]
  },
  "extensions": {
    "server_latency": {
      "parsing_ns": 51146,
      "processing_ns": 288051,
      "encoding_ns": 1067559
    },
    "txn": {
      "start_ts": 63,
      "lin_read": {
        "ids": {
          "1": 6
        }
      }
    }
  }
}

- Expected behaviour and actual result.

I would expect the nodes returned to be identical to the unordered case except for the ordering.

**Comment by peterstace** [comment]
Thank you for reporting, given your detailed instructions this was very easy to reproduce. I'll look at making a fix.

**Comment by peterstace** [comment]
Minimum example that exhibits the problem:

```
name: string @index(exact) .
```
```
_:michael <name> "Michael" .
_:michael <friend> _:sang .

_:sang <name> "상현"@ko .
_:sang <name> "Sang Hyun"@en .
```
```
{
  michael_friends_first(func: eq(name, "Michael")) {
    name
    friend (orderdesc: name@.) {
      name@en
    }
  }
}
```

It gives:
```{
  "michael_friends_first": [
    {
      "name": "Michael",
      "friend": [
        {
          "name@en": "Sang Hyun"
        },
        {
          "name@en": "Sang Hyun"
        }
      ]
    }
  ]
}
```
But should give:
```
{
  "michael_friends_first": [
    {
      "name": "Michael",
      "friend": [
        {
          "name@en": "Sang Hyun"
        }
      ]
    }
  ]
}
```

There seems to be a bad interaction between multiple languages and sorting.

**Comment by peterstace** [comment]
Merged to master in ed85541
